tests/e2e/tests/configure-and-run-analysis.test.ts

```diff
@@ -35,10 +35,131 @@ test.describe(`Configure extension and run analysis`, () => {
     await vscodeApp.runAnalysis();
     await vscodeApp.waitDefault();
     await expect(vscodeApp.getWindow().getByText('Analysis completed').first()).toBeVisible({
-      timeout: 300000,
+      timeout: 400000,
     });
   });
 
+  const collator = new Intl.Collator(undefined, { sensitivity: 'base', numeric: true });
+  const isAscending = (arr: string[]) =>
+    arr.every((v, i, a) => i === 0 || collator.compare(a[i - 1], v) <= 0);
+
+  /**
+   * Checks that the issues list is sorted correctly in both ascending and descending order.
+   * Verifies that the descending order is not ascending, and ascending order is sorted.
+   */
+  test('Set list kind and sort (Issues ascending and descending)', async () => {
+    await vscodeApp.setListKindAndSort('issues', 'descending');
+    const names_descending = await vscodeApp.getListNames('issues');
+    //console.log('Issue names:', names_descending);
+    expect(isAscending(names_descending)).toBe(false);
+
+    await vscodeApp.setListKindAndSort('issues', 'ascending');
+    const names_ascending = await vscodeApp.getListNames('issues');
+    //console.log('Issue names:', names_ascending);
+    expect(isAscending(names_ascending)).toBe(true);
+  });
+
+  /**
+   * Checks that each file name in the provided list is valid.
+   * A valid file name has a known bare name or a file extension.
+   * @param names - The list of file names to validate.
+   */
+  function expectFilesListLooksValid(names: string[]) {
+    const knownBare = new Set([
+      'README',
+      'LICENSE',
+      'Makefile',
+      'Dockerfile',
+      'BUILD',
+      'WORKSPACE',
+    ]);
+    for (const full of names) {
+      const base = full.split(/[\\/]/).filter(Boolean).pop() ?? full;
+
+      const hasDotExt = /\.[A-Za-z0-9._-]+$/.test(base);
+      const isKnownBare = knownBare.has(base);
+      expect(
+        hasDotExt || isKnownBare,
+        `Not a file-like name: "${full}" (base: "${base}")`
+      ).toBeTruthy();
+    }
+  }
+
+  /**
+   * Checks that the files list is sorted correctly in both ascending and descending order.
+   * Also validates that all file names look valid.
+   */
+  test('Set list kind and sort (Files ascending and descending)', async () => {
+    await vscodeApp.setListKindAndSort('files', 'ascending');
+    const files_ascending = await vscodeApp.getListNames('files');
+    expectFilesListLooksValid(files_ascending);
+    const asc = [...files_ascending].sort((a, b) => collator.compare(a, b));
+    //console.log('Issue names:', files_ascending);
+    expect(files_ascending).toEqual(asc);
```

Comment:
Can't we use the `isAscending` function you defined above?

Reply:
I wrote this method before I added **isAscending** function and forgot to update.